### DISCUSSION TITLE
Set meta description for a specific route

### DISCUSSION THREAD

**Discussion by justoverclockl** [laravel/framework#38008] [General]
Hi guys, i'm here to ask some help, 

i have:
```
 ->route('/my-page', 'namespace/my-app'),
 ->content(function (Document $document) {
            $document->head[] = '<meta name="Description" CONTENT="TRANSLATION HERE">';}),
```

i need to add that only to a specific route (/my-page)
how can i achieve this?

**Comment by usamamuneerchaudhary** [comment]
The way i do is, add the meta in the headers, and use in my specific route's blades templates, like so:
`@section('description','contenthere')`
and in the header:
`<meta name="description" content="@yield('description')"/>`

I think that's the better solution instead of manipulating the content headers on the go.

**Reply by justoverclockl** [reply]
can u show me a little code example? i need to learn that

**Reply by usamamuneerchaudhary** [reply]
Let's say you've a `app.blade.php` file which has the basic skeleton like 
```<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
<title>
        @yield('title')
</title>
<meta name="description" content="@yield('description')"/>
.....
</head>
......
</html>
```

Now in your blade file, you're extending this `app.blade.php`, 
```
@extends('app')
@section('title', 'my page title')
@section('description','my page description')
```

This way you'll have different sets of meta tags for each page in your application. Hope this makes sense to you now :)

**Reply by justoverclockl** [reply]
Thank you very much!!
